Missing type annotation in `strawberry.fastapi.BaseContext` causes mypy to trip

## Describe the Bug

I built a custom context based on [the guide in the docs](https://strawberry.rocks/docs/guides/authentication):

```python
from strawberry.fastapi import BaseContext

class CustomContext(BaseContext):
    @property
    def user(self) -> User:
        ...

async def get_context() -> CustomContext:
    return CustomContext()
```

With that I receive the following mypy error:
```shell
 error: Call to untyped function "CustomContext" in typed context  [no-untyped-call]
```

For now, I added the following workaround to my code:
```python
class CustomContext(BaseContext):
    if typing.TYPE_CHECKING:
        def __init__(self) -> None:
            pass
    
    ...
```

## System Information

 - Operating system: macOS Monterey
 - Strawberry version (if applicable): `0.158.1`
 - FastAPI version (if applicable): `0.92.0`
 - mypy version (if applicable): `0.991` (also tested with `1.0.1`)

## Additional Context

I'm happy to provide a PR to address the issue myself.

--- a/RELEASE.md
+++ b/RELEASE.md
@@ -1,0 +1,4 @@
+Release type: patch
+
+Add a type annotation to `strawberry.fastapi.BaseContext`'s `__init__` method so that
+it can be used without `mypy` raising an error.

--- a/strawberry/fastapi/context.py
+++ b/strawberry/fastapi/context.py
@@ -14,7 +14,7 @@ MergedContext = Union[
 class BaseContext:
     connection_params: Optional[Any] = None
 
-    def __init__(self):
+    def __init__(self) -> None:
         self.request: Optional[Union[Request, WebSocket]] = None
         self.background_tasks: Optional[BackgroundTasks] = None
         self.response: Optional[Response] = None

